### ISSUE TITLE
Display only appropriate reportlets based on campaign type

### ISSUE THREAD

**Issue by JoeMurray** [systopia/de.systopia.campaign#2]
For campaigns that don't involve revenue, don't show those related to revenue or KPIs related to ROI.

For campaigns with other KPI, like open rates for a email blast, or number of responses for an SMS blast, display appropriate reportlets for those campaign types. 

More generally, provide infrastructure that will allow this to be configured for each type of campaign.

**Comment by bjendres** [comment]
> For campaigns that don't involve revenue, don't show those related to revenue or KPIs related to ROI.

This is going to be difficult, because we don't know which campaign types are supposed to yield revenue. What we could do, however, is to hide those when they currently don't _have_ any.

> More generally, provide infrastructure that will allow this to be configured for each type of campaign.

There is. As described in the https://github.com/systopia/de.systopia.campaign/blob/master/USERDOC.md#implementing-custom-kpis you can implement a hook to manipulate (or remove) the KPIs to your liking.

**Comment by michau** [comment]
@systopia @JoeMurray Regarding hiding selected KPIs that currently don't have any contributions (ergo, don't yeld revenue), would below logic be ok?

**If "Number of Contributions (completed)" equals 0 and "Total Revenue Goal" equals 0 or null, hide whole KPIs section and render a message asking to define "Total Revenue Goal" or register some contributions against this campaign.**

Whaddya think?

**Comment by bjendres** [comment]
I'm not sure that _everyone_ would agree with that, so my suggestion would be:
- implement the hook in an extra extension (you might have to do that anyway)
- have it remove all KPIs in the above case
- add logic to the KPI panel in _this_ extension to hide if no KPIs provided.

This way we don't break it for people who do want to see empty KPIs

What do you think?

**Comment by michau** [comment]
Fair enough. Our few cases consistently involve hiding financial data for non-financial campaigns, so from my little ship this landscape seems like whole world. ;-)

Thanks for hints - we are indeed working on extension to extension (https://bitbucket.org/caltha/pl.infozmiana.campaigngoodies) which will involve some additional logic and KPIs. Once we have it ready and tested (few weeks), we can potentially plan for contributing some of the features/KPIs back to original extension - if they will be of any use for you.

**Comment by bjendres** [comment]
Sounds good! Looking forward to your input and feedback!
